doc/api/child_process.markdown

```diff
@@ -729,10 +729,30 @@ an error but may have unforeseen consequences. Specifically, if the process
 identifier (PID) has been reassigned to another process, the signal will be
 delivered to that process instead which can have unexpected results.
 
+See `kill(2)` for general reference.
+
 Note that while the function is called `kill`, the signal delivered to the
-child process may not actually terminate the process.
+child process may not actually terminate the process. E.g. on `linux` child
+processes of child processes will not be terminated when attempting to kill
+their parent. This is likely to happen when running a new process in a shell
+or with use of the `shell` option of `ChildProcess`.
 
-See `kill(2)`
+```js
+'use strict';
+const spawn = require('child_process').spawn;
+
+let child = spawn('sh', ['-c',
+  `node -e "setInterval(() => {
+      console.log(process.pid + 'is alive')
+    }, 500);"`
+  ], {
+    stdio: ['inherit', 'inherit', 'inherit']
+  });
+
+setTimeout(() => {
+  child.kill(); // does not terminate the node process in the shell
```

Comment:
Unfortunately there currently isn't much you can do about this:

You need to keep references to the grand children's pid and kill it directly. Unfortunately that isn't very easy in such a scenario. If you are lucky you can assume that the pid is the child's pid +1. 

I am currently writing (or rather researching) a libuv addition to query all children's pids of a process. This is not to straightforward, since e.g. on OS X there is no C function for that.

The alternative is to detach and just orphan the grandchild